python/py-igraph/Portfile

```diff
@@ -39,26 +40,35 @@ if {${name} ne ${subport}} {
                             port:bison \
                             port:flex
 
+    variant external_glpk description {Build with external GLPK} {        
+        depends_lib-append      port:glpk
+    }
+
+    variant external_linalg description {Build with external BLAS, LAPACK, ARPACK} {
+        depends_lib-append      port:lapack port:arpack
+    }
+
+    default_variants        +external_glpk
+
     set extra_configure_args { }
 
-    variant external_glpk description {Build with external GLPK} {
+    if {[variant_isset external_glpk]} {
         lappend extra_configure_args    --with-external-glpk
-        depends_lib-append              port:glpk
     }
 
-    variant external_linalg description {Build with external BLAS, LAPACK, ARPACK} {
+    if {[variant_isset external_linalg]} {
         lappend extra_configure_args    --with-external-blas --with-external-lapack --with-external-arpack
-        depends_lib-append              port:lapack port:arpack
     }
 
-    default_variants        +external_glpk
+    build.env-append        IGRAPH_EXTRA_CONFIGURE_ARGS=[join $extra_configure_args]
 
-    build.env-append        IGRAPH_EXTRA_CONFIGURE_ARGS="[join $extra_configure_args]"
+    pre-test {
```

Comment:
Does anything actually run during pre-test? Everything appears to be during the normal test phase.

Reply:
I took this from other Python ports (see e.g. [py-blosc](https://github.com/macports/macports-ports/blob/master/python/py-blosc/Portfile#L50)). I'm not sure why, but if I don't put setting `PYTHONPATH` in `pre-test`, it won't be picked up by the time `test.run` is actually run.

When submitting the original py-igraph 0.8.3 update, I did not notice that things weren't working because I ran `port test` only after the port was installed. It was loading the installed version of the package. However, `port test` should work even if the package is not installed, and it should pick up the version of the package that was just built. After placing `test.env` in `pre-test`, this works correctly now.